cypress-io/cypress
# fix: issue-7715

- Closes #7715

### User facing changelog
In the terminal, users will now see the source map line and column number in the error stack trace. Fixes #7715

### Additional details

Previously, we were finding the stack trace with the source map -- but we weren't doing anything with it 😱 it was appended to the error as `error.sourceMappedStack` and left to linger, perhaps for debugging purposes.

This PR reassigns the `stack` property to the value of the source mapped stack and removes the lone property `sourceMappedStack`. 

The purists among us may say,
> Hey! Should we really be manipulating the `stack` property? 

To which I will answer, 
> We were already manipulating the `stack` property by pointing it to the user's invocation stack via [preferredStackAndCodeFrameIndex](https://github.com/cypress-io/cypress/blob/develop/packages/driver/src/cypress/error_utils.ts#L434). This change just ensures the right line number/column is printed in the already-manipulated stack.

### Steps to test
1. Create a dummy test that will fail, i.e. in `packages/driver/cypress/e2e/commands/agents.cy.js`
```
Cypress.env('retries', 0)

it.only('should fail', () => {
  expect(false).to.be.true
    // OR
    // cy.get('invalid')
    // 	.should('be.visible')
    // OR
    // cy.origin('http://foobar.com:3500')
})
```
2. From the project root, start the runner in one tab with `cd packages/runner && yarn watch`
3. From the project root, run your test in another tab with `yarn cypress:run --headed --project packages/driver --spec packages/driver/cypress/e2e/commands/agents.cy.js --browser chrome --no-exit`
4. Confirm that the error stack line number/column in the terminal matches the value in VS Code
5. Confirm that the error stack line number/column in the GUI matches the line number/column in VS Code

**For example, all three of these show line 8, column 25: `cypress/e2e/commands/agents.cy.js:8:25`**
- VS Code:
<img width="400" alt="Screen Shot 2022-08-26 at 1 02 31 PM" src="https://user-images.githubusercontent.com/26726429/186983226-2c16b020-2113-4d38-9d2d-6a824c876120.png">

- Terminal:

<img width="500" alt="Screen Shot 2022-08-26 at 1 02 15 PM" src="https://user-images.githubusercontent.com/26726429/186983212-e0934dbe-8712-4716-89f8-cbd3f08f4e07.png">

- GUI:

<img width="400" alt="Screen Shot 2022-08-26 at 1 07 39 PM" src="https://user-images.githubusercontent.com/26726429/186983182-e7d5d499-1291-4606-af14-f6b560f0ae43.png">


### How has the user experience changed?
Now we see the correct line number/column in the stack trace. Note: you'll notice the stack trace link has changed from `http://localhost:3500/__cypress/tests?p=cypress/e2e/commands/agents.cy.js:X:XX` to `webpack:///./cypress/e2e/commands/agents.cy.js:X:XX`, but this has no impact on the user experience since the localhost domain was not relevant to the user's source map in the codebase. In fact, we may be interested in stripping it as we do in the CodeFrame source map line

**Before:**
<img width="700" alt="Screen Shot 2022-08-26 at 1 18 23 PM" src="https://user-images.githubusercontent.com/26726429/186984480-0e89f5da-c11f-426b-98ea-f2511fd0ea24.png">

**After:**
<img width="608" alt="Screen Shot 2022-08-26 at 1 19 15 PM" src="https://user-images.githubusercontent.com/26726429/186984649-6d01f161-c78c-43aa-b63e-c76f24dcd323.png">


### PR Tasks
